### PR TITLE
Closes #48, Closes #50; Rename inline to value, allow soft-keywords to be used as names, recognize functional interfaces, add regressions and tests.

### LINKED ISSUE #48: Incorrect highlighting of functional interfaces and its member.
Self explanatory with the following example:

```kotlin
fun interface ITestInterface {
    fun sam(): Unit
}
```

### LINKED ISSUE #50: data is highlighted as a class modifier in different scenarios
It seems that `data` is always highlighted as a class modifier i.e. as in `data class`.

```kotlin
fun foo(data: String) {}
val data = "42"
foo(data = data)

data class Klass(val data: Int)
Klass(data = 42)
```

### PR DESCRIPTION
**Pull request by Animeshz**
Fixes #48.
Fixes #50.

### CODE REVIEW

File: src/keywords.YAML-tmLanguage [hunk 1]
```diff
@@ -17,14 +17,23 @@ repository:
                 match: \b(\_)\b
                 name: punctuation.definition.variable.kotlin
             -
-                match: \b(data|inline|tailrec|operator|infix|typealias|reified)\b
+                match: \b(tailrec|operator|infix|typealias|reified|copy(?=\s+fun|\s+var))\b
```
Comment (nishtahir) [inline src/keywords.YAML-tmLanguage:20]: Mind adding a test case covering this?

File: src/keywords.YAML-tmLanguage [hunk 2]
```diff
@@ -8,7 +8,7 @@ repository:
                 match: \b(in|is|as|assert)\b
                 name: keyword.operator.kotlin
             -
-                match: \b(const)\b
+                match: \b(const|lateinit)\b
```
Comment (nishtahir) [inline src/keywords.YAML-tmLanguage:11]: 👍

Comment (Animeshz) [reply src/keywords.YAML-tmLanguage:11]: Oops, I just figured out `const` and `lateinit` both can also be used as variables, i.e. are soft keywords 😛 should've been with their friends!

**Comment by Animeshz** [comment]
Any updates?

**Review by nishtahir — APPROVED** [review]

**Comment by nishtahir** [comment]
> Any updates?

Sorry for the late review - I was out on vacation when this came in

**Pull request merged**